influxdb/client.py

```diff
@@ -254,25 +243,21 @@ def request(self, url, method='GET', params=None, data=None,
                 break
             except (requests.exceptions.ConnectionError,
                     requests.exceptions.HTTPError,
-                    requests.exceptions.Timeout) as _e:
-                _error = _e
+                    requests.exceptions.Timeout):
                 _try += 1
                 if self._retries != 0:
                     retry = _try < self._retries
                 if method == "POST":
                     time.sleep((2 ** _try) * random.random() / 100.0)
-        if _error:
-            raise(_error)
+                if not retry:
+                    raise
```

Comment:
linters like this approach a little better, and it also gives more digestible tracebacks IMO:

```python
>>> e = None
>>> try:
...     raise ValueError('this is a raised value error')
... except ValueError as _e:
...     e = _e
... 
>>> raise(e)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 2, in <module>
ValueError: this is a raised value error
```

```python
>>> try:
...     raise ValueError('this is a raised value error')
... except ValueError:
...     raise
... 
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: this is a raised value error
```

This seems to be the original intent.